shell/command.go

```diff
@@ -122,22 +123,31 @@ func (c *Command) Run() (Summary, string, error) {
 			return summary, "", err
 		}
 	}
+
+	// handle command errors
 	errors := &bytes.Buffer{}
+
 	// send error output to buffer & stderr
 	if stderr != nil {
 		stderrOutput := c.Stderr
 		if stderrOutput == nil {
 			stderrOutput = os.Stderr
 		}
 		err = c.ScanStderr(stderr, stderrOutput, errors)
-		if err != nil {
-			return summary, errors.String(), err
+
+		if e := cmd.Wait(); err == nil {
+			err = e
```

Comment:
Might be more correct to change the logic and always set err when e != nil. 
When cmd.Wait() returns an error it should be returned to get the error code of the process.
Had done it this way to preserve the original logic for error reporting (but in the original code, cmd.Wait was not called in this case)

Reply:
I see what you mean, if there's an error reading from the pipe it's going to end the process and probably killing restic before the end.

That is a very unlikely event but I definitely overlooked it. You're right it's better to wait until the end of the command before reporting the error 👍🏻

Reply:
Ok. Will update it.